<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="omav_hook_visualization">

    <!-- Properties -->
    <xacro:property name="manipulator_length" value="0.62"/>   <!-- [m] -->
    <xacro:property name="manipulator_radius" value="0.0075"/>  <!-- [m] -->

    <xacro:property name="tool_length" value="0.05"/>         <!-- [m] -->
    <xacro:property name="tool_radius" value="0.0075"/>         <!-- [m] -->

    <xacro:property name="mass_ouzel" value="5.0"/>
    <xacro:property name="ixx_ouzel" value="0.12"/>
    <xacro:property name="iyy_ouzel" value="0.14"/>
    <xacro:property name="izz_ouzel" value="0.18"/>
    <xacro:property name="radius_ouzel" value="0.2"/>
    <xacro:property name="height_ouzel" value="0.3"/>

    <link name="odom_omav"/>

    <joint name="odom_omav_to_omav_base" type="fixed">
        <parent link="odom_omav"/>
        <child link="omav_base"/>
        <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
    </joint>

    <link name="omav_base">
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <cylinder radius="${radius_ouzel}" length="${height_ouzel}"/>
            </geometry>
        </collision>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <cylinder radius="${radius_ouzel}" length="${height_ouzel}"/>
            </geometry>
            <material name="">
                <color rgba="1 1 1 1.0"/>
            </material>
        </visual>
    </link>

    <joint name="omav_to_rod" type="fixed">
        <parent link="omav_base"/>
        <child link="rod"/>
        <origin rpy="$(arg omav_to_rod_angle) 0 0" xyz="${manipulator_length/2} 0.0 0.0"/>
    </joint>

    <link name="rod">
        <collision>
            <origin rpy="0 1.5708 0" xyz="0 0 0"/>
            <geometry>
                <cylinder radius="${manipulator_radius}" length="${manipulator_length}"/>
            </geometry>
        </collision>
        <visual>
            <origin rpy="0 1.5708 0" xyz="0 0 0"/>
            <geometry>
                <cylinder radius="${manipulator_radius}" length="${manipulator_length}"/>
            </geometry>
            <material name="">
                <color rgba="1 1 1 1"/>
            </material>
        </visual>
    </link>

    <joint name="rod_to_tip" type="fixed">
        <parent link="rod"/>
        <child link="tip"/>
        <origin rpy="0 0 0" xyz="${manipulator_length/2} 0.0 0.0"/>
    </joint>
    <!--
        <joint name="tip_to_thick" type="fixed">
            <parent link="tip"/>
            <child link="thick"/>
            <origin rpy="0 0 0" xyz="-0.04 0.0 0.0"/>
        </joint>

        <link name="thick">
            <collision>
                <origin rpy="0 1.5708 0" xyz="0 0 0"/>
                <geometry>
                    <cylinder radius="0.015" length="0.095"/>
                </geometry>
                <material name="">
                    <contact name="rubber"/>
                </material>
            </collision>
            <visual>
                <origin rpy="0 1.5708 0" xyz="0 0 0"/>
                <geometry>
                    <cylinder radius="0.015" length="0.095"/>
                </geometry>
                <material name="">
                    <color rgba="1 1 1 1"/>
                </material>
            </visual>
        </link>
-->
        <link name="tip">
            <collision>
                <origin rpy="0 0 0" xyz="0 0 0"/>
                <geometry>
                    <sphere radius="${tool_radius}"/>
                </geometry>
                <material name="">
                    <contact name="rubber"/>
                </material>
            </collision>
            <visual>
                <origin rpy="0 0 0" xyz="0 0.0 0"/>
                <geometry>
                    <sphere radius="${tool_radius}"/>
                </geometry>
                <material name="">
                    <color rgba="1 1 1 1"/>
                </material>
            </visual>
        </link>

        <joint name="tip_to_hook" type="fixed">
            <parent link="tip"/>
            <child link="hook"/>
            <origin rpy="0 0 0" xyz="0.0 0.0 -${tool_length/2}"/>
        </joint>

        <link name="hook">
            <collision>
                <origin rpy="0 0 0" xyz="0 0 0"/>
                <geometry>
                    <cylinder radius="${tool_radius}" length="${tool_length}"/>
                </geometry>
                <material name="">
                    <contact name="rubber"/>
                </material>
            </collision>
            <visual>
                <origin rpy="0 0 0" xyz="0 0 0"/>
                <geometry>
                    <cylinder radius="${tool_radius}" length="${tool_length}"/>
                </geometry>
                <material name="">
                    <color rgba="1 1 1 1"/>
                </material>
            </visual>
        </link>

    </robot>